### PR TITLE
Add --oplog parameter to mongodump

### PR DESCRIPTION
**Pull request by jacklagare**

### CODE REVIEW

**Review by AlbertoPdRF — CHANGES_REQUESTED** [review]
Thanks for following up on #334 @jacklagare! 🧐

I'm afraid though that this as is won't work in production, as the `DB_URL` environment variable includes the database. Maybe we could do some string manipulation with it?

Also, I think that `lint-staged` is not working for you, you'll have to delete the `node_modules` folder and re-run `npm install` before committing again for the code style issues to get automatically fixed.

By the way, we have a Slack workspace that you can join if you want, so we can address small details quicker, you can find an invite link on the README!

And just for the record, this is related to #27.

**Comment by jacklagare** [comment]
@AlbertoPdRF, I implemented parsing of the MongoDB URI. Kindly check.

**Comment by AlbertoPdRF** [comment]
I'll test this out later today!

File: api/db/backup.js [hunk 1]
```diff
@@ -1,8 +1,13 @@
 const exec = require('child_process').exec;
 const path = require('path');
+const parser = require('mongodb-uri');
```
Comment (AlbertoPdRF) [inline api/db/backup.js]:
```suggestion
const mongodbUri = require('mongodb-uri');
```

File: api/db/backup.js [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
 const { getDbUrl } = require('./connection');
 
 const createBackup = () => {
+  // Remove database name from URI to allow full database dumps
+  const uri = parser.parse(getDbUrl());
+  const mongoDbUrl = `${uri['scheme']}://${uri['username']}:${uri['password']}@${uri['hosts'][0]['host']}/`;
```
Comment (AlbertoPdRF) [inline api/db/backup.js]:
```suggestion
  // Remove database (and options) from URI to allow full database dumps
  const parsedUri = mongodbUri.parse(getDbUrl());
  delete parsedUri.database;
  delete parsedUri.options;
  const uri = mongodbUri.format(parsedUri);
```

File: api/db/backup.js [hunk 3]
```diff
@@ -11,7 +16,7 @@ const createBackup = () => {
     now.getMonth() + 1
   }${now.getDate()}.gz`;
 
-  const dumpCmd = `mongodump --uri='${getDbUrl()}' --gzip --archive='${dumpsPath}/${dumpFilename}'`;
+  const dumpCmd = `mongodump --uri='${mongoDbUrl}' --gzip --archive='${dumpsPath}/${dumpFilename}' --oplog`;
```
Comment (AlbertoPdRF) [inline api/db/backup.js]:
```suggestion
  const dumpCmd = `mongodump --uri='${uri}' --gzip --archive='${dumpsPath}/${dumpFilename}' --oplog`;
```

**Review by AlbertoPdRF — CHANGES_REQUESTED** [review]
I've left a few code suggestions, mostly about naming but also taking advantage of the new dependency to avoid crafting the URI manually.

Also, please note that there's a conflict left to resolve on the `package-lock.json` file.

I haven't yet tested this on a production-like environment, I'll do that as soon as I can and report the results back here!

File: api/db/backup.js [hunk 4]
```diff
@@ -1,8 +1,16 @@
 const exec = require('child_process').exec;
 const path = require('path');
+const mongodbUri = require('mongodb-uri');
 const { getDbUrl } = require('./connection');
+const { parse } = require('path');
```
Comment (AlbertoPdRF) [inline api/db/backup.js]:
```suggestion
```
Oops, this line should go.

**Comment by jacklagare** [comment]
Done. Those were awesome updates. I didn't know you can do that.

**Comment by AlbertoPdRF** [comment]
Ok, I've now tested this and it works as expected! :tada: Thanks so much for addressing this @jacklagare!

Note: as mentioned on Slack, before merging this I'd like to get rid of the unrelated changes currently present on the `package-lock.json` file though. Let me know if you need help with that!

**Review by AlbertoPdRF — APPROVED** [review]
Looks clean now, thanks again for your work on this @jacklagare! 🙌

**Pull request merged**